### ISSUE TITLE
About 10 % memory allocation per process when terra is loaded?

### ISSUE THREAD

**Issue by DavorJ** [rspatial/terra#1657]
I am on Windows 10/11 and observe large memory reservation by terra_1.7.83 / R_4.3.3.

To reproduce it is very simple: Make a new project with an `.Rprofile` in the root containing the following piece of code:

```R
loadNamespace("terra")
``` 

Now on ´R´ command line, simply make a cluster with `cluster <- parallel::makeCluster(20)` and observe the memory allocation. When each clusters spins up, it will execute `loadNamespace("terra")`. This call seems to reserve about 10 % of the available physical RAM per process. 

From:
![image](https://github.com/user-attachments/assets/f42317c7-01d1-4d30-9679-cf8633a13eba)

To:
![image](https://github.com/user-attachments/assets/6a96db80-5bb1-4059-9d31-9ec76535786f)

This is 56GB (63-7) just for loading terra in 20 parallel processes. Is this intended (?), and if so, how can it be adjusted? If terra was the only thing used in the code, then I wouldn't mind, but I my case I have lots of other memory-intensive code that would also require some memory, and terra seems to be reserving too much.  If the system cannot reserve enough memory, then you end up with:

``
Error in inDL(x, as.logical(local), as.logical(now), ...) : unable to load shared object '[lib_path]/library/terra/libs/x64/terra.dll': LoadLibrary failure: The paging file is too small for this operation to complete.
Calls: loadNamespace -> library.dynam -> dyn.load -> inDL
Execution halted
`` 

The memory reservation seems to be related to GDAL loading in [`R/zzz.R`](https://github.com/rspatial/terra/blob/master/R/zzz.R). I have tried playing a bit with config settings, but to no avail:

```R
options(terra_default = list(
  memfrac = 0.9,
  memmin = 0.01,
  memmax = 0.01 # GB, to limit terra memory usage.
))

Sys.setenv("GDAL_CACHEMAX"="32MB")
Sys.setenv("GDAL_MAX_DATASET_POOL_RAM_USAGE"="64MB")
Sys.setenv("VSI_CACHE"="FALSE")

loadNamespace("terra")

```

Any suggestions?

<details>Versions

```
> packageVersion("terra")
[1] '1.7.83'
> terra::gdal(lib="all")
    gdal     proj     geos
 "3.8.2"  "9.3.1" "3.11.2"
```
</details>

**Comment by arthurgailes** [comment]
Related to: https://github.com/r-spatial/sf/issues/2517. I've found that downgrading to 4.2.1 works, YMMV.

**Comment by DavorJ** [comment]
Good point yes. I confirm `sf_1.0-16`/`terra_1.7-71` don't exhibit this issue on `R4.2.3` . So something in `R4.3` release is the culprit?

**Comment by arthurgailes** [comment]
Yes. Rtools 43 upgrades to GDAL 3.6, which I believe is the source of the problem. [GDAL 3.6](https://github.com/OSGeo/gdal/blob/v3.6.1/NEWS.md) sets GDAL_CACHEMAX to 5% of memory, which I *think* is the issue. This also is cumulative if loading 2 spatial packages.

I haven't been able to figure out a way to fix it. Setting the relevant R environment variables seems to register, but not to have any affect.

**Comment by rhijmans** [comment]
It seems to me that setting GDAL_CACHEMAX to 5% happened a long time ago, in [GDAL 2.1](https://github.com/OSGeo/gdal/blob/v3.6.1/NEWS.md#gdal-210---overview-of-changes)

**Comment by arthurgailes** [comment]
Yes, you're right. Then I'm back to square one on the exact cause.
